includes/class-wc-cart-totals.php

```diff
@@ -247,15 +275,38 @@ protected function get_fees_from_cart() {
 			$fee->taxable   = $fee->object->taxable;
 			$fee->total     = wc_add_number_precision_deep( $fee->object->amount );
 
-			if ( $this->calculate_tax && $fee->object->taxable ) {
-				$fee->taxes     = WC_Tax::calc_tax( $fee->total, WC_Tax::get_rates( $fee->object->tax_class, $this->cart->get_customer() ), false );
-				$fee->total_tax = array_sum( $fee->taxes );
+			if ( $this->calculate_tax ) {
+				if ( 0 > $fee->total ) {
+					// Negative fees should have the taxes split between all items so it works as a true discount.
+					$tax_class_costs = $this->get_tax_class_costs( $order );
```

Comment:
`<b>Notice</b>:  Undefined variable: order in <b>/srv/www/wordpress-default/public_html/wp-content/plugins/woocommerce/includes/class-wc-cart-totals.php</b> on line <b>281</b><br />`